src/Dependencies/Updater.php

```diff
@@ -41,63 +27,74 @@ private function updateDependency($dep)
 
         $type = explode('/', $dep)[0];
         $name = explode('/', $dep)[1];
+
+        $result = $this->prepareForUpdate();
+        if ($result->isErr()) {
+            return $result;
+        }
+
         $depArray = $this->jsonFile->getDependency($type, $name);
         if ($depArray === []) {
             return \Result\Result::err('No matching dependency in whippet.json');
         }
 
-        $this->updateHash();
-        $this->loadGitignore();
+        return $this->update([$type=>[$depArray]]);
+    }
 
-        echo sprintf("[Updating %s/%s]\n", $type, $depArray['name']);
-        $result = $this->addDependencyToLockfile($type, $depArray);
+    public function updateAll()
+    {
+        $result = $this->prepareForUpdate();
```

Comment:
Shame we can't factor this repeated `prepareForUpdate()` out too. What's the dependency on the order of the different bits of it? Could any of it be moved into `update()`?

Looks like maybe `loadWhippetFiles` needs to be repeated but `updateHash();` and `loadGitignore();` could happen inside `update()`?